R/PackageData2Csv.R

```diff
@@ -0,0 +1,89 @@
+#' Create a CSV variant of .rda files
+#'
+#' The function should be run with a path to a package directory.
+#' It will then look through the `data` directory of the package,
+#' and for all data sets that are data frames, create CSV variants
+#' in a `data-csv` directory.
+#'
+#' @param pkg The R package where we'd like to generate CSVs of any
+#'   data frames.
+#' @param dir A character string representing the path to the folder.
+#'   where the CSV files should be written. If no such directory
+#'   exists, one will be created (recursively).
+#' @param overwrite Boolean to indicate if to overwrite any existing
+#'   files that have conflicting names in the directory specified.
+#' @export
+#' @examples
+#'
+#' \dontrun{
+#'   PackageData2Csv("openintro")
+#'   list.files("data-csv")
+#' }
+PackageData2Csv <- function(pkg, dir = "data-csv", overwrite = FALSE) {
```

Comment:
Can we call it something like `write_pkg_data()`? I know the functions in this package have ThisCase styling but I've been trying to stick to snake_case.

Reply:
> Can we call it something like `write_pkg_data()`? I know the functions in this package have ThisCase styling but I've been trying to stick to snake_case.

Sure. My rationale had been to avoid mixing styles, but I see that's already happening in the package.

Right now the assortment of naming styles looks/feels messy. I know I'd started the inconsistency in how I did camelCase for functions (originally camelCase, then I later used CamelCase). We probably should refactor things to create a consistent style, though if we're moving to snake_case, I don't look forward to that work of converting textbook files.